pkg/master/reconcilers/lease.go

```diff
@@ -224,7 +220,13 @@ func (r *leaseEndpointReconciler) doReconcile(serviceName string, endpointPorts
 	}
 
 	glog.Warningf("Resetting endpoints for master service %q to %v", serviceName, masterIPs)
-	_, _, err = r.endpointStorage.Update(ctx, e.Name, rest.DefaultUpdatedObjectInfo(e), rest.ValidateAllObjectFunc, rest.ValidateAllObjectUpdateFunc, false, &metav1.UpdateOptions{})
+	if shouldCreate {
+		if _, err = r.endpointClient.Endpoints(corev1.NamespaceDefault).Create(e); errors.IsAlreadyExists(err) {
+			err = nil
```

Comment:
Technically we've lost an update here, I believe.  I think it's safe because (1) we expect the racing updater to have done the same thing and (2) we'll reconcile it on the next polling cycle.  (But if we ever go to a watching controller, I think 2 would cease to hold)